### PR TITLE
Lasso algorithm

### PR DESCRIPTION
**Pull request by jdematias**
First Lasso basic version implementation

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,4 +2,4 @@ language: scala
 jdk:
   - oraclejdk8
 sudo: false
-script: mvn clean verify
\ No newline at end of file
+before_install: sh scripts/build-flink-ps.sh
```
Comment (VenturaDelMonte) [inline .travis.yml]: i think you rebased on the wrong file as you can see [here](https://github.com/proteus-h2020/proteus-solma/blob/master/.travis.yml).

File: scripts/build-flink-ps.sh [hunk 2]
```diff
@@ -2,16 +2,26 @@
```
Comment (VenturaDelMonte) [inline scripts/build-flink-ps.sh]: this file should not be modified by this PR.

File: src/main/scala/eu/proteus/solma/lasso/Lasso.scala [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  implicit def predictImplementation[K <: Lasso.OptionLabeledVector] = {
+    new LassoPredictOperation[K]
+  }
+
+  implicit def transformImplementation[T <: Lasso.OptionLabeledVector] = {
+    new LassoStreamTransformOperation[T]
+  }
+}
+
+class Lasso extends StreamPredictor[Lasso] with Estimator[Lasso]{
```
Comment (VenturaDelMonte) [inline src/main/scala/eu/proteus/solma/lasso/Lasso.scala]: Estimator is for batch. Please, make sure you use StreamEstimator.

File: src/main/scala/eu/proteus/solma/lasso/LassoParameterServer.scala [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+    * Limit of unanswered pulls at a worker instance.
+    * @param iterationWaitTime
+    * Time to wait for new messages at worker. If set to 0, the job will run infinitely.
+    * PS is implemented with a Flink iteration, and Flink does not know when the iteration finishes,
+    * so this is how the job can finish.
+    * @return
+    * Stream of Parameter Server model updates and predicted values.
+    */
+  def transformLasso(model: Option[DataStream[(Int, LassoModel)]] = None)
+                    (inputSource: DataStream[OptionLabeledVector],
```
Comment (VenturaDelMonte) [inline src/main/scala/eu/proteus/solma/lasso/LassoParameterServer.scala]: Solma features [StreamEvent](https://github.com/proteus-h2020/proteus-solma/blob/master/src/main/scala/eu/proteus/solma/events/StreamEvent.scala) as main data object as we wanted to support proteus scenario (i.e., a stream event contains a sparse vector). Do you think we could eventually integrate StreamEvent and LabeledVector?

**Comment by VenturaDelMonte** [comment]
Hi @jdematias, please, let me know when you are done with resolving all the comments. Please, could you also take care of the conflicting files?

**Comment by jdematias** [comment]
Sorry @VenturaDelMonte I have closed this PR and open other without conflicting and with your requested changes https://github.com/proteus-h2020/proteus-solma/pull/35

**Pull request closed without merging**